src/components/common/navbar/index.tsx

```diff
@@ -0,0 +1,69 @@
+import React, { useEffect } from "react";
+import { Icon } from "@components/common";
+import {
+  openMenu,
+  closeMenu,
+  setActiveLink,
+} from "@/state/features/menu/menuSlice";
+
+import { Link, useLocation } from "react-router-dom";
+import { useAppDispatch, useAppSelector } from "@/state/app/hooks";
+
+import styles from "./Navbar.module.css";
+
+const Navbar = () => {
+  const dispatch = useAppDispatch();
+  const { menuOpen, activeLink } = useAppSelector((state) => state.menu);
+
+  const location = useLocation();
+
+  const handleMenu = (status: boolean) => {
+    status ? dispatch(openMenu()) : dispatch(closeMenu());
+  };
+
+  const handleActiveLink = (activeLink: string) => {
+    dispatch(setActiveLink(activeLink));
+    dispatch(closeMenu());
+  };
+
+  const navLinks = [
+    { href: "/about", name: "About" },
+    //I'm not including "Contact" here as it's a Modal and it's not supposed to change url path
+    //However, array structure is made in case more links are needed in the future
+  ];
+
+  useEffect(() => {
+    location.pathname === "/about" && activeLink !== "/about"
+      ? dispatch(setActiveLink("About"))
+      : dispatch(setActiveLink(""));
+  }, []);
```

Comment:
I think `navLinks` could be outside `Navbar` function and also it would be better if you work with some `contants` or something like that to avoid typo error and also it will be better if you need to change at anytime. I mean don't use '/about' and 'About' as string. WDYT

Reply:
Like it! End up adding them as an `enum` with an utils function to change from name to path